### PR TITLE
Update API to V4

### PR DESCRIPTION
**Pull request by dfernandezramos**
Since some models affect different test classes I've done a single PR separating each endpoint group/endpoint tests class by commits.

### CODE REVIEW

**Comment by dfernandezramos** [comment]
- Failing tests in master are still failing but I've checked that my changes don't affect them.
- GetChampionMasteryAsyncTest () on ChampionMasteryTests has a commented Assert line since Riot is returning wrong data (check difference between v3 and v4). The behavior should be the same but the id encryption is wrong or something like that.
- I've changed some tests requests since old requests were asking for old data. I've updated them.
- Since summoners id's and accountId's are encrypted now depending on the API key which has made the request, I've made the tests that need Id's make a previous request by summoner name to get the encrypted data.

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs [hunk 1]
```diff
@@ -10,26 +10,37 @@ namespace RiotNet.Tests
...
+            var championMastery = await client.GetChampionMasteryAsync(encryptedSummonerId, 412L); // Thresh
 
             Assert.That(championMastery, Is.Not.Null);
             Assert.That(championMastery.ChampionId, Is.EqualTo(412L));
             Assert.That(championMastery.LastPlayTime.Kind, Is.EqualTo(DateTimeKind.Utc));
             Assert.That(championMastery.LastPlayTime, Is.GreaterThan(new DateTime(2015, 1, 1, 0, 0, 0, DateTimeKind.Utc)).And.LessThanOrEqualTo(DateTime.UtcNow));
             Assert.That(championMastery.ChampionLevel, Is.AtLeast(1), "Invalid champion level.");
             Assert.That(championMastery.ChampionPoints, Is.AtLeast(1), "Invalid number of champion points.");
-            Assert.That(championMastery.PlayerId, Is.EqualTo(34172230L));
+            //Assert.That(championMastery.SummonerId, Is.EqualTo(EncryptedSummonerId));
```
Comment (aj-r) [inline RiotNet.Tests/Apis/ChampionMasteryTests.cs:37]: Add a comment explaining why this line is commented out

File: RiotNet.Tests/Apis/TournamentTests.cs [hunk 2]
```diff
@@ -13,6 +13,47 @@ namespace RiotNet.Tests
     [TestFixture]
     public class TournamentTests : TestBase
     {
+        string rndmInternetMan_Id;
```
Comment (aj-r) [inline RiotNet.Tests/Apis/TournamentTests.cs]: I would prefer that variable names don't contain underscores (`rndmInternetManId` is fine)

File: RiotNet/Models/Summoner.cs [hunk 3]
```diff
@@ -14,12 +14,17 @@ public class Summoner
         /// </summary>
         [Key]
         [DatabaseGenerated(DatabaseGeneratedOption.None)]
-        public long Id { get; set; }
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the player universally unique identifier (PUUID)
+        /// </summary>
+        public string PUUID { get; set; }
```
Comment (aj-r) [inline RiotNet/Models/Summoner.cs]: Only capitalize the first letter of acronyms (`Puuid`)

**Review by aj-r — CHANGES_REQUESTED** [review]
Looks great! Just a few minor things.

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs (hunk 1 above)
Comment (dfernandezramos) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs:37]: done

File: RiotNet.Tests/Apis/TournamentTests.cs (hunk 2 above)
Comment (dfernandezramos) [reply RiotNet.Tests/Apis/TournamentTests.cs]: done

Comment (dfernandezramos) [reply RiotNet.Tests/Apis/TournamentTests.cs]: done

File: RiotNet/Models/Summoner.cs (hunk 3 above)
Comment (dfernandezramos) [reply RiotNet/Models/Summoner.cs]: done

**Comment by dfernandezramos** [comment]
Changes processed. I'll merge the fixup commits with each proper commit after approve 👍

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs [hunk 4]
```diff
@@ -10,26 +10,38 @@ namespace RiotNet.Tests
...
+            var championMastery = await client.GetChampionMasteryAsync(encryptedSummonerId, 412L); // Thresh
 
             Assert.That(championMastery, Is.Not.Null);
             Assert.That(championMastery.ChampionId, Is.EqualTo(412L));
             Assert.That(championMastery.LastPlayTime.Kind, Is.EqualTo(DateTimeKind.Utc));
             Assert.That(championMastery.LastPlayTime, Is.GreaterThan(new DateTime(2015, 1, 1, 0, 0, 0, DateTimeKind.Utc)).And.LessThanOrEqualTo(DateTime.UtcNow));
             Assert.That(championMastery.ChampionLevel, Is.AtLeast(1), "Invalid champion level.");
             Assert.That(championMastery.ChampionPoints, Is.AtLeast(1), "Invalid number of champion points.");
-            Assert.That(championMastery.PlayerId, Is.EqualTo(34172230L));
+            // Riot is returning wrong data for no knwown reason. It returns the summoner id encripted in a different way than the other endpoints. Check this when v4 is totally active and v3 deprectated
```
Comment (aj-r) [inline RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Typo: known

**Review by aj-r — CHANGES_REQUESTED** [review]

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs (hunk 4 above)
Comment (aj-r) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Also: encrypted

Comment (dfernandezramos) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Done

**Comment by dfernandezramos** [comment]
All changes processed. Since your last changes request were two typos, I've already merged the fixup commits with their respective ones. 👍

**Review by aj-r — APPROVED** [review]

**Comment by dfernandezramos** [comment]
Please let me know if you need something else. I assume you will merge this PR when you can.

**Comment by aj-r** [comment]
Everything is fine, I just wanted to double-check that you were merging to the right branch first (which I can't do on my phone for some reason). Looks good!

**Pull request merged**